### PR TITLE
Add config to enable mixed types as string in GpuJsonToStruct & GpuJsonScan

### PR DESCRIPTION
**Pull request by andygrove**
Closes https://github.com/NVIDIA/spark-rapids/issues/10253

## Changes in this PR:

- Add repro test for `A mix of lists and structs within the same column is not supported`
- Enable "mixed types as strings" support, fixing this test
- Add a new config to enable this, because there are some known issues (https://github.com/rapidsai/cudf/issues/14864)

### CODE REVIEW

**Comment by andygrove** [comment]
build

**Comment by andygrove** [comment]
build

File: sql-plugin/src/main/scala/org/apache/spark/sql/catalyst/json/rapids/GpuReadJsonFileFormat.scala
Comment (jlowe) [inline sql-plugin/src/main/scala/org/apache/spark/sql/catalyst/json/rapids/GpuReadJsonFileFormat.scala:1]: 2024 copyrights

File: sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuJsonToStructs.scala
Comment (jlowe) [inline sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuJsonToStructs.scala:1]: 2024 copyrights

**Review by jlowe — APPROVED** [review]
This looks OK, but we probably want to change the headline as it implies this is now enabled when it's actually disabled by default.

**Comment by jlowe** [comment]
build

**Comment by andygrove** [comment]
Test failures with `st_hash_grpby_sum`

```
BlockManagerMaster: Failed to remove shuffle 11 - An executor with RapidsShuffleManager is trying to use a ShuffleBufferCatalog that isn't initialized.
2024-01-26T19:11:39.8073853Z [2024-01-26T19:09:27.685Z] java.lang.IllegalStateException: An executor with RapidsShuffleManager is trying to use a ShuffleBufferCatalog that isn't initialized.

```

**Comment by andygrove** [comment]
build failing due to https://github.com/NVIDIA/spark-rapids/issues/10291

**Review by revans2 — APPROVED** [review]

**Comment by andygrove** [comment]
build

**Pull request merged**